dnf/conf/substitutions.py

```diff
@@ -45,35 +45,6 @@ def _update_from_env(self):
             elif key in numericvars:
                 self[key] = val
 
-    @staticmethod
-    def _split_releasever(releasever):
-        # type: (str) -> tuple[str, str]
-        pos = releasever.find(".")
-        if pos == -1:
-            releasever_major = releasever
-            releasever_minor = ""
-        else:
-            releasever_major = releasever[:pos]
-            releasever_minor = releasever[pos + 1:]
-        return releasever_major, releasever_minor
-
-    def __setitem__(self, key, value):
-        if Substitutions.is_read_only(key):
-            raise ReadOnlyVariableError(f"Variable \"{key}\" is read-only", variable_name=key)
```

Comment:
If I understand correctly removal of this `__setitem__` is causing the problem you mentioned from the `doc/examples/install_extension.py` example for API users.

Why can't the change happen here to be more compatible?
I think all that is technically needed is to remove the read only check no?

(It could also use the `releasever_major, releasever_minor = libdnf.conf.ConfigParser.splitReleasever(value)` here).

Reply:
Old callers using `detect_releasever` will still have the releasever `"10.1"` split to `releasever_major = "10"; releasever_minor = "1"`, that happens in the `@releasever.setter` in `dnf/conf/config.py`.

The reason callers should now use `detect_releasevers` is because otherwise they wouldn't detect distribution overrides of `releasever_major` and `releasever_minor`. e.g. if releasever = `10` but the distribution overrides `releasever_minor = "0"`.

Reply:
Oh, yes, I see what you mean now, the example is setting `base.conf.substitutions['releasever']` directly. Yes, I can rearrange things to avoid breaking that.

Reply:
I moved the splitting logic back to `Substitutions`.